gengokit/httptransport/httptransport.go

```diff
@@ -239,6 +259,28 @@ func createDecodeConvertFunc(f Field) string {
 	case strings.Contains(f.GoType, "string"):
 		fType = "%s := %s"
 	}
+	if !f.IsBaseType || f.Repeated {
```

Comment:
well, I didn't put it in it's own function, but I have modified it extensively to be much more clear what it's doing, with comments explaining why. It may still benefit from being in a function, but let me know what you think of the improvements I've made so far.